### PR TITLE
Improve vaccination centers import script.

### PR DESCRIPTION
**Pull request by n1k0**
Trello cards: 

- https://trello.com/c/bsowPANj/540-g%C3%A9rer-la-suppression-dans-le-cas-dun-r%C3%A9import-o%C3%B9-un-erp-nest-plus-r%C3%A9f%C3%A9renc%C3%A9-le-mettre-hors-ligne
- https://trello.com/c/zyUMh02V/539-exclure-limport-des-centres-d%C3%A9clar%C3%A9s-ferm%C3%A9s
- https://trello.com/c/DtB3mmyK/538-exclure-les-centres-o%C3%B9-la-prise-de-rv-est-impossible-pour-la-plupart-de-nos-utilisateurs-car-r%C3%A9serv%C3%A9s-aux-personnels-soignants

### CODE REVIEW

**Comment by n1k0** [comment]
@celiacheff moved the import logic to a record mapper object so it's *kinda* testable, even if db is unavoidably involved, so tests are kinda slow. But I think it's a bit cleaner a much safer that way. I could go further but we have to land this asap, so... Thanks!

File: erp/mapper/vaccination.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        if self.erp_exists and self.erp.has_accessibilite():
+            accessibilite = self.erp.accessibilite
+        else:
+            accessibilite = Accessibilite(erp=self.erp)
+        accessibilite.commentaire = commentaire
+        accessibilite.save()
+
+        return self.erp
+
+    def __build_commentaire(self):
```
Comment (celiacheff) [inline erp/mapper/vaccination.py:80]: I'd rather use single underscore for "privates" methods, see also: https://stackoverflow.com/a/6930223

File: erp/mapper/vaccination.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        "c_nom": "nom",
+        "c_adr_num": "numero",
+        "c_adr_voie": "voie",
+        "c_com_nom": "commune",
+        "c_com_cp": "code_postal",
+        "c_com_insee": "code_insee",
+        "c_rdv_tel": "telephone",
+    }
+
+    RESERVE_PS = [
```
Comment (celiacheff) [inline erp/mapper/vaccination.py:21]: "PS" is not obvious, maybe change it to "PROS" or "PROFESSIONALS"

File: tests/core/test_text.py [hunk 3]
```diff
@@ -35,3 +35,8 @@ def test_ucfirst():
     assert text.ucfirst("") == ""
     assert text.ucfirst("a") == "A"
     assert text.ucfirst("foo") == "Foo"
+
+
+def test_strip_if_str():
+    assert text.strip_if_str(None) is None
+    assert text.strip_if_str(" foo ") == "foo"
```
Comment (celiacheff) [inline tests/core/test_text.py:42]: Not sure about the added value of this test, looks like testing internals

File: tests/erp/mapper/test_vaccination.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    sample_closed["properties"]["c_date_fermeture"] = "2021-01-01"
+
+    m = RecordMapper(sample_closed, today=datetime(2021, 1, 2))
+    with pytest.raises(RuntimeError) as err:
+        m.process(activite_cdv)
+
+    assert "SKIPPED: Centre fermé le 2021-01-01" in str(err.value)
+
+
+def test_skip_importing_reserve_ps(activite_cdv, neufchateau, sample_record_ok):
```
Comment (celiacheff) [inline tests/erp/mapper/test_vaccination.py:52]: Same as above, using "pros" instead of "ps" would be more explicit imho

**Review by celiacheff — APPROVED** [review]
Nice refactoring. Ok for me.

File: erp/mapper/vaccination.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (n1k0) [reply erp/mapper/vaccination.py:80]: I don't know... In my tests:

```python
In [8]: class Foo:
   ...:     def __plop(self):
   ...:         print("yeah")
   ...:     def plop(self):
   ...:         self.__plop()
   ...:
   ...:
   ...:

In [9]: Foo().plop()
yeah

In [10]: Foo().__plop()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-e656aedd56a0> in <module>
----> 1 Foo().__plop()

AttributeError: 'Foo' object has no attribute '__plop'
```

While using a single undercore:

```python
In [11]: class Foo:
    ...:     def _plop(self):
    ...:         print("yeah")
    ...:     def plop(self):
    ...:         self._plop()
    ...:
    ...:
    ...:
    ...:

In [12]: Foo().plop()
yeah

In [13]: Foo()._plop()
yeah
```

I like the double underscore convention really hiding the method from the public interface.

Thoughts?

File: tests/core/test_text.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (n1k0) [reply tests/core/test_text.py:42]: It's part of the public API of `core.lib.text` so there's total value of testing it.

File: tests/erp/mapper/test_vaccination.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (n1k0) [reply tests/erp/mapper/test_vaccination.py]: Addressed in https://github.com/MTES-MCT/acceslibre/pull/136/commits/71bc108f68b7438bafd66b7e206b0167e3ee0452

File: erp/mapper/vaccination.py (hunk 2 above)
Comment (n1k0) [reply erp/mapper/vaccination.py]: Addressed in https://github.com/MTES-MCT/acceslibre/pull/136/commits/71bc108f68b7438bafd66b7e206b0167e3ee0452

File: tests/erp/mapper/test_vaccination.py (hunk 4 above)
Comment (n1k0) [reply tests/erp/mapper/test_vaccination.py]: Addressed in https://github.com/MTES-MCT/acceslibre/pull/136/commits/71bc108f68b7438bafd66b7e206b0167e3ee0452

File: erp/mapper/vaccination.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (n1k0) [reply erp/mapper/vaccination.py]: Okay the SO argument makes sense, addressed in https://github.com/MTES-MCT/acceslibre/pull/136/commits/6f0fae649683da7157f9f77b71d41ac14b0f0f94

File: erp/mapper/vaccination.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (n1k0) [reply erp/mapper/vaccination.py]: Adressed in https://github.com/MTES-MCT/acceslibre/pull/136/commits/71bc108f68b7438bafd66b7e206b0167e3ee0452

**Pull request merged**